src/Workspaces/Remote/ServiceHub/Host/RemoteWorkspace.cs

```diff
@@ -408,5 +381,46 @@ public ValueTask<(Solution solution, bool updated)> TryUpdateWorkspaceCurrentSol
                 return tuple.solution;
             }
         }
+
+        private sealed class LazySolution : IAsyncDisposable, IDisposable
+        {
+            private readonly RemoteWorkspace _remoteWorkspace;
+            private readonly Checksum _solutionChecksum;
+            private readonly CancellationTokenSource _cancellationTokenSource = new();
+            private readonly Task<Solution> _task;
+
+            public LazySolution(RemoteWorkspace remoteWorkspace, Checksum solutionChecksum, Func<CancellationToken, Task<Solution>> getSolutionAsync)
```

Comment:
instead of passing in RemoteWorkspace+Checksum, we could pass an async callback to be called when the final dispose actually happens.  that would make this helper nicely generalized and independent of RemoteWorkspace.

Reply:
Now done this way